nibabel/funcs.py

```diff
@@ -122,8 +124,13 @@ def concat_images(images, check_affines=True):
         if check_affines:
             if not np.all(img.affine == affine):
                 raise ValueError('Affines do not match')
-        out_data[i] = img.get_data()
-    out_data = np.rollaxis(out_data, 0, len(i0shape)+1)
+        out_data[i] = img.get_data().copy()
+    if axis is not None:
+        out_data = np.concatenate(out_data, axis=axis)
```

Comment:
@matthew-brett For some reason, I had it in my head that what we actually had was a list.  I'm shocked that the few tests I did worked at all.

I'll start over on this.